registry/application/services/validate_demo_component_service.py

```diff
@@ -74,3 +75,22 @@ def extract_file_details_from_file_path(path):
             raise Exception(msg)
         path_values = path.split('/')
         return path_values[0], path_values[2], path_values[4]
+
+    @staticmethod
+    def update_demo_component_build_status(org_uuid, service_uuid, build_status):
+        status = "SUCCEEDED" if build_status else "FAILED"
+        service = service_repo.get_service_for_given_service_uuid(org_uuid=org_uuid, service_uuid=service_uuid)
+        if service:
+            assets = service.assets
+            demo_details = assets['demo_files']
+            demo_details.update({'status': status})
+            assets.update({'demo_files': demo_details})
+        else:
+            raise Exception(f"Service {service_uuid} not found for org {org_uuid}")
+        service_repo.update_service_assets(org_uuid=org_uuid, service_uuid=service_uuid, assets=assets)
+        if status == "SUCCEEDED":
```

Comment:
If build status is FAILED we should update the service state to CHANGE_REQUESTED